### PR TITLE
add flattening and minifying options to config view

### PR DESCRIPTION
**Pull request by deads2k**
This adds `--minify` and `--flatten` to `kubectl config view`.  

`--minify` displays only those pieces of the kubeconfig that are relevant to the `current-context`.

`--flatten` chases file references and displays the contents as though the kubeconfig files contained the data and not the references.

Neither command actually modifies any kubeconfig file, it simply controls what is displayed.  Because the display is now done using the external packages, it can output to a file and later used a kubeconfig in its own right.

@jlowdermilk This builds on previous pulls `kubectl config`

@derekwaynecarr We discussed doing this to make the injection of kubeconfigs via secrets more manageable.

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
@deads2k - awesome!  this is a big help in unblocking https://github.com/GoogleCloudPlatform/kubernetes/issues/4997  I will try this out today and give feedback.

File: docs/kubectl-config-view.md [hunk 1]
```diff
@@ -29,8 +29,10 @@ $ kubectl config view -o template --template='{{range .users}}{{ if eq .name "e2
 ### Options
 
 ```
+      --flatten=false: flatten the resulting kubeconfig file into self contained output (useful for making secrets)
```
Comment (derekwaynecarr) [inline docs/kubectl-config-view.md]: I know it's useful for making secrets, but I think its just generally useful for making self-contained kubeconfig files that can be copied across machines independent of the secret use case.

File: docs/kubectl-config-view.md [hunk 2]
```diff
@@ -29,8 +29,10 @@ $ kubectl config view -o template --template='{{range .users}}{{ if eq .name "e2
...
   -h, --help=false: help for view
       --merge=true: merge together the full hierarchy of kubeconfig files
+      --minify=false: remove all information not used by current-context from the output (useful for making secrets)
```
Comment (derekwaynecarr) [inline docs/kubectl-config-view.md]: Agree its useful for secrets, but I am not sure if alluding to it in help text says how.  I would just leave the ( ... ) out of the help

File: pkg/client/clientcmd/api/helpers.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+			return err
+		}
+
+		*path = ""
+	}
+
+	return nil
+}
+
+// ResolvePaths updates the given refs to be absolute paths, relative to the given base directory
```
Comment (derekwaynecarr) [inline pkg/client/clientcmd/api/helpers.go]: ResolvePath

File: pkg/client/clientcmd/api/helpers.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
```
Comment (jlowdermilk) [inline pkg/client/clientcmd/api/helpers.go]: 2015

**Comment by deads2k** [comment]
comments addressed.

File: pkg/client/clientcmd/api/helpers_test.go [hunk 5]
```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
```
Comment (jlowdermilk) [inline pkg/client/clientcmd/api/helpers_test.go]: 2015 here too

Comment (deads2k) [inline pkg/client/clientcmd/api/helpers_test.go]: grrr.  ::starts looking at https://github.com/GoogleCloudPlatform/kubernetes/blob/master/hack/verify-boilerplate.sh::

**Comment by derekwaynecarr** [comment]
Can you squash commits and are you able to run gendocs locally without issue?

**Comment by deads2k** [comment]
Squashed and fixed up gendocs.

**Comment by derekwaynecarr** [comment]
LGTM

**Pull request merged**